internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/issuecreator/JIRAIssueCreator.java

```diff
@@ -0,0 +1,242 @@
+     *
+     * @param result result string.
+     * @return True if vulnerability exists in report else return False.
+     */
+    private boolean isVulnerabilityExistInReport(String result) {
+        JSONArray resposeArray = new JSONArray(result);
+        boolean isVulnerabilityExistInReport = false;
+        if (resposeArray.length() > 0) {
+            isVulnerabilityExistInReport = true;
+        }
+        return isVulnerabilityExistInReport;
+    }
+
+    /**
+     * Create JIRA new ticket.
+     *
+     * @param product  product name.
+     * @param assignee mail address of a person who is responsible for this ticket.
+     * @return Created issue data.
+     */
+    private String createNewTicket(String product, String assignee, String summary) {
+
+        String createIssueData = "{\"fields\": {\"project\": {\"key\":\"" + projectKey + "\"}," +
+                "\"summary\":" + "\"" + summary + "\"" + ",  \"assignee\": {\"name\": \"" +
+                assignee + "\"}," + "\"labels\":" + "[\"" + issueLabel + "\"]" + "," +
+                "\"description\":" + "\"" + product + "\"" + "," +
+                "\"issuetype\":{\"name\":\"" + issueType + "\"}}}";
+        return createIssueData;
+    }
+
+    /**
+     * Generating the comment that has to be added with the attachment
+     *
+     * @return comment that needs to be added when uploading a file
+     */
+    private String createComment(String product) {
+
+        String comment;
+        DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
+        Date date = new Date();
+        comment = "{\"body\": \"This Report is generated for " + product + " on " + dateFormat.format(date) + ".\"}";
+        return comment;
+    }
+
+    /**
+     * Checking if there is any issues already reported in the Jira with the samw topic
+     *
+     * @param auth    Base64 encoded authorization paramters
+     * @param summary JIRA heading which used to find the JIRA existance
+     * @return returning the JIRA key, if already jira is created else return an empty String
+     */
+
+    private String checkForIssueExistence(String auth, String summary, String projectKey) {
+
+        String responseIssuue;
+        String key = "";
+        JSONObject availableIssue = null;
+        summary = summary.replace("[", "");
+        summary = summary.replace("]", "");
+
+        try {
+            String issueExistenceCheckUrl = IssueCreatorConstants.WSO2_JIRA_BASE_URL +
+                    "rest/api/2/search?jql=project+%3d+" + projectKey +
+                    "+AND+text+%7e+%22" + summary + "%22" + "&fields=" + "";
+            responseIssuue = jira.invokeGetMethod(auth, issueExistenceCheckUrl);
+            availableIssue = new JSONObject(responseIssuue);
+
+        } catch (AuthenticationException e) {
+            log.error("Authentication failed", e);
+        }
+
+        try {
+            assert availableIssue != null;
```

Comment:
Why are we using assertions here? they are disabled by default.

Reply:
Resolved